Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM ubuntu:20.04 as builder
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+# Timezone data is needed during the installation of dependencies,
+# since cmake depends on the tzdata package. In an interactive terminal,
+# the user selects the timezone at installation time. Since this needs
+# to be a non-interactive terminal, we need to setup some sort of default.
+# The UTC one seemed the most suitable one.
+RUN echo 'tzdata tzdata/Areas select Etc' | debconf-set-selections; \
+    echo 'tzdata tzdata/Zones/Etc select UTC' | debconf-set-selections; \
+    apt-get update && apt-get upgrade -y && apt-get install --no-install-recommends -y \
+    cmake \
+    ninja-build \
+    gcc \
+    g++ \
+    maven \
+    flex \
+    bison \
+    libxml2-utils \
+    dpkg-dev
+COPY . /tmp/cbmc
+WORKDIR /tmp/cbmc
+RUN cmake -S . -Bbuild -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_CXX_COMPILER=/usr/bin/g++ && cd build; ninja -j2
```

Comment:
I don't think we need the `-DCMAKE_*_COMPILER` settings as those will be no different from the system defaults. And `ninja -C build -j2` would spare you the `cd build`.

Reply:
Or `cmake --build build`

Reply:
Hm, when it comes to the `WORKDIR` I think it might be better to actually keep it. From the [docker documentation](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers) it seems there's no longer any massive need to avoid extraneous instructions, as only a few introduce new layers, and I think it's cleaner this way.

What do you think about this?